fury/ui/elements.py

```diff
@@ -3052,3 +3055,395 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
+            hyp = np.hypot(size[0], size[1])
+            if self.max_size and hyp > self.max_size:
+                hyp = self.max_size
+            self.shape.outer_radius = hyp
+
+        self.cal_bounding_box(self.position)
+
+    def left_button_pressed(self, i_ren, _obj, shape):
+        mode = self.drawpanel.current_mode
+        if mode == "selection":
+            click_pos = np.array(i_ren.event.position)
+            self._drag_offset = click_pos - self.position
+            i_ren.event.abort()
+        elif mode == "delete":
+            self._scene.rm(self.shape.actor)
+            i_ren.force_render()
+        else:
+            self.drawpanel.left_button_pressed(i_ren, _obj, self.drawpanel)
+
+    def left_button_dragged(self, i_ren, _obj, shape):
+        if self.drawpanel.current_mode == "selection":
+            if self._drag_offset is not None:
+                click_position = i_ren.event.position
+                relative_canvas_position = click_position - self._drag_offset - self.drawpanel.position
+                new_position = self.clamp_position(relative_canvas_position)
+                self.drawpanel.canvas.update_element(self, new_position)
+            i_ren.force_render()
+        else:
+            self.drawpanel.left_button_dragged(i_ren, _obj, self.drawpanel)
+
+
+class DrawPanel(UI):
+    """The main Canvas(Panel2D) on which everything would be drawn.
+    """
+
+    def __init__(self, size=(400, 400), position=(0, 0), is_draggable=False):
+        """Init this UI element.
+
+        Parameters
+        ----------
+        size : (int, int), optional
+            Width and height in pixels of this UI component.
+        position : (float, float), optional
+            (x, y) in pixels.
+        is_draggable : bool, optional
+            Whether the background canvas will be draggble or not.
+        """
+        self.panel_size = size
+        super(DrawPanel, self).__init__(position)
+        self.is_draggable = is_draggable
+        self.current_mode = None
+
+        if is_draggable:
+            self.current_mode = "selection"
+
+        self.shape_list = []
+
+    def _setup(self):
+        """Setup this UI component.
+
+        Create a Canvas(Panel2D).
+        """
+        self.canvas = Panel2D(size=self.panel_size)
+        self.canvas.background.on_left_mouse_button_pressed = self.left_button_pressed
+        self.canvas.background.on_left_mouse_button_dragged = self.left_button_dragged
+
+        mode_data = {
+            "selection": ["selection.png", "selection-pressed.png"],
+            "line": ["line.png", "line-pressed.png"],
+            "quad": ["quad.png", "quad-pressed.png"],
+            "circle": ["circle.png", "circle-pressed.png"],
+            "delete": ["delete.png", "delete-pressed.png"]
+        }
+
+        padding = 5
+        mode_panel_size = (len(mode_data) * 35 + 2 * padding, 40)
+        self.mode_panel = Panel2D(size=mode_panel_size, color=(0.5, 0.5, 0.5))
```

Comment:
You can include the fourth variable for color 'rgba'.

Reply:
The alpha value is explicitly set using the `opacity` parameter for all the UI